ddtrace/contrib/trace_utils.py

```diff
@@ -545,6 +545,17 @@ def activate_distributed_headers(tracer, int_config=None, request_headers=None,
     if override is False:
         return None
 
+    # Only extract and activate if we don't already have an activate span
+    # DEV: Only do this if there is an active Span, an active Context is fine to override
+    current_span = tracer.current_span()
+    if current_span:
+        log.debug(
+            "will not extract distributed headers, a Span(trace_id%d, span_id=%d) is already active",
+            current_span.trace_id,
+            current_span.span_id,
+        )
+        return
+
```

Comment:
```suggestion
```

Reply:
this is the main value we are trying to drive with this change, avoid extracting context at all in conditions where we don't need to.